src/SmartComponents/RunTaskModal/RunTaskModal.js

```diff
@@ -1,53 +1,35 @@
 import React, { useEffect, useState } from 'react';
-import {
-  Alert,
-  Button,
-  Flex,
-  FlexItem,
-  Form,
-  FormGroup,
-  Modal,
-  TextInput,
-  ValidatedOptions,
-} from '@patternfly/react-core';
+import { Button } from '@patternfly/react-core/dist/js/components/Button';
+import { Modal } from '@patternfly/react-core/dist/js/components/Modal';
 import propTypes from 'prop-types';
-import SystemTable from '../SystemTable/SystemTable';
 import { ExclamationCircleIcon } from '@patternfly/react-icons';
-import {
-  AVAILABLE_TASKS_ROOT,
-  INFO_ALERT_SYSTEMS,
-  TASKS_API_ROOT,
-  TASK_ERROR,
-} from '../../constants';
-import { fetchSystems } from '../../../api';
+import { TASK_ERROR } from '../../constants';
 import EmptyStateDisplay from '../../PresentationalComponents/EmptyStateDisplay/EmptyStateDisplay';
 import ExecuteTaskButton from '../../PresentationalComponents/ExecuteTaskButton/ExecuteTaskButton';
-import ReactMarkdown from 'react-markdown';
 import { dispatchNotification } from '../../Utilities/Dispatcher';
 import { EXECUTE_TASK_NOTIFICATION } from '../../constants';
 import { isError } from '../completedTaskDetailsHelpers';
-import warningConstants from '../warningConstants';
+import SystemsSelect from './SystemsSelect';
+import InputParameters from './InputParameters';
 
 const RunTaskModal = ({
   description,
   error,
   isOpen,
   name,
+  parameters,
   selectedSystems,
   setIsRunTaskAgain,
   setModalOpened,
   slug,
   title,
 }) => {
   const [selectedIds, setSelectedIds] = useState(selectedSystems);
-  const [taskName, setTaskName] = useState();
+  const [taskName, setTaskName] = useState(name);
```

Comment:
Why does the taskName and title need to be in a state?

Reply:
`taskName` is a unique name given to the specific running task, whereas the title is the general name of the task. Think:
```
{
  title: 'dog',
  name: 'fido'
}
```

If no name is given, then the name will default to the `title`. So we use this state to update the specific name of the task.